packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-metric-action.ts

```diff
@@ -0,0 +1,80 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for CloudWatch metric.
+ */
+export interface CloudWatchMetricActionProps extends CommonActionProps {
+  /**
+   * The CloudWatch metric name.
+   *
+   * Supports substitution templates.
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+   */
+  readonly metricName: string,
+
+  /**
+   * The CloudWatch metric namespace name.
+   *
+   * Supports substitution templates.
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+   */
+  readonly metricNamespace: string,
+
+  /**
+   * A string that contains the timestamp, expressed in seconds in Unix epoch time.
+   *
+   * Supports substitution templates.
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+   *
+   * @default - none -- Defaults to the current Unix epoch time.
+   */
+  readonly metricTimestamp?: string,
+
+  /**
+   * The metric unit supported by CloudWatch.
+   *
+   * Supports substitution templates.
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+   */
+  readonly metricUnit: string,
+
+  /**
+   * A string that contains the CloudWatch metric value.
+   *
+   * Supports substitution templates.
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+   */
+  readonly metricValue: string,
+}
```

Comment:
Can't we use the [`Metric` class](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-cloudwatch.Metric.html) from the CloudWatch module instead of all these separate props here?

Reply:
@skinny85 
I thought it too. But if using `Metric` class, we will not be able to use substitution templates.
So I think it is better that we provide `fromMetric()` method.

Reply:
Interesting... what's the problem that prevents using `Metric` with substitution templates?

Reply:
@skinny85 
I had envisioned using `Metric` class as following.
```ts
const metric = new cloudwatch.Metric({
  namespace: 'test-namespace',
  metricName: 'test-name',
  unit: cloudwatch.Unit.COUNT,
});
new CloudWatchPutMetricAction(metric, {
  metricValue: '${temperature}',
  metricTimestamp: '${timestamp}',
});
```

In above case, I predicted that users might think "I cannot use substitution templates in `namespace`, `metricName` and `unit`".

Hmmm🤔. As I was writing this, I realized that there was nothing preventing from using substitution templates in properties besides `unit` as following.

```ts
const metric = new cloudwatch.Metric({
  namespace: '${namespace}',
  metricName: '${topic(2)}',
});
```

I'm a little uncomfortable with it, in that substitution templates are written in `Metric` class props. How about?

And `unit` cannot be used with substitution templates because `unit` is enum in `Metric`.

Reply:
How common it is to use a dynamic Unit, do you think?

We can work around it by having something like:

```ts
new CloudWatchPutMetricAction(metric, {
  dynamicUnit: `${sth}`,
});
```

It _is_ a little bit awkward, I agree.

Reply:
I think it's a pretty rare case that a `unit` is used dynamically.

For just one thing, How about this?
We keep current props, and we provide `metric()` method like [`metric-filter.ts`](https://github.com/yamatatsu/aws-cdk/blob/master/packages/%40aws-cdk/aws-logs/lib/metric-filter.ts#L56-L63).
Because, in many case, `CloudWatchPutMetricAction` will generate new custom metric for example temperature, weight and moisture. And alerms will be created with this metrics.
So I think users maybe will want metric after they make the topic rules.

Reply:
OK, sounds good.